src/utils/sources-tree/getDirectories.js

```diff
@@ -9,48 +9,42 @@ import { getURL } from "./getURL";
 import type { TreeNode, TreeDirectory } from "./types";
 import type { Source } from "../../types";
 
-function findSource(sourceTree: TreeDirectory, sourceUrl: string): TreeNode {
-  let returnTarget = null;
-  function _traverse(subtree: TreeNode) {
-    if (subtree.type === "directory") {
-      for (const child of subtree.contents) {
-        _traverse(child);
-      }
-    } else if (!returnTarget) {
-      if (subtree.path.replace(/http(s)?:\//, "") == sourceUrl) {
-        returnTarget = subtree;
-        return;
-      }
+function _traverse(subtree: TreeNode, source: Source) {
+  if (subtree.type === "source") {
+    if (subtree.contents.id === source.id) {
```

Comment:
I thought we would be able to do`subtree.contents === source`, but alas when we  blackbox a source we create a new reference...